### PR TITLE
Buildkite Deployments

### PR DESCRIPTION
**Pull request by aabrook**
Setting up the deployment pipeline to auto deploy from buildkite

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -17,3 +17,19 @@ steps:
...
     agents:
       queue: elastic
+
+  - name: ":rocket: Release staging"
+    branches: "master"
+    type: "manual"
+
+  - name: Build and deploy to staging
+    branches: "master"
+    command: "bin/buildkite bin/ci deploy staging"
```
Comment (aabrook) [inline .buildkite/pipeline.yml]: These need to be run bin/deploy

File: bin/deploy.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+const tokenPresent = (token) => !!process.env[token]
+const tokenNotPresent = (token) => !tokenPresent(token)
+
+const assertEnvVars = () => {
+  const requiredEnvVars = [
+    'BUCKET',
+    'BUCKET_PREFIX',
+    'JEFFREY_TOKEN'
+  ]
```
Comment (aabrook) [inline bin/deploy.js]: I think I want this list external so the script can be used across projects

File: bin/deploy [hunk 3]
```diff
@@ -7,7 +7,7 @@ export JEFFREY_TOKEN=$JEFFREY_TOKEN_STAGING
 export BUCKET=$BUCKET_STAGING
 export BUCKET_PREFIX=$PREFIX_STAGING
 
-if [ $DEPLOY_ENV == "production" ];
+if [ $DEPLOY_ENV == "production" ]; then
```
Comment (kizza) [inline bin/deploy]: Several of the changes in this commit are in fact !fixups of previous commits :shrug: #firstworldproblems :)

File: bin/buildkite [hunk 4]
```diff
@@ -20,8 +20,12 @@ export COMMIT="${BUILDKITE_COMMIT}"
 export HEALTH_ENDPOINT='https://nps-collect.everydayhero.io/health'
 export JOB_ID="${BUILDKITE_JOB_ID}"
 export PUSH_TO_QUAY=false
-export JEFFREY_TOKEN_STAGING="${JEFFREY_TOKEN_STAGING}"
-export JEFFREY_TOKEN_PRODUCTION="${JEFFREY_TOKEN_PRODUCTION}"
+export JEFFREY_TOKEN_STAGING="${JEFFREY_TOKEN_STAGING:-}"
```
Comment (kizza) [inline bin/buildkite]: Again, unsure - but i'm not used to seeing step related env variables within `buildkite` or `localkite` but rather within the related step (perhaps there is a reason here?)   I'm used to only seeing `plain-pipeline` related vars such as https://github.com/everydayhero/kanga/blob/master/bin/buildkite

**Pull request merged**